src/stuff.m4

```diff
@@ -76,11 +76,13 @@ dnl The argbash script generator will pick it up and (re)generate that one as we
 dnl
 dnl $1: the filename (assuming that it is in the same directory as the script)
 dnl $2: what has been passed to DEFINE_SCRIPT_DIR as the first param
+dnl TODO: add shellcheck source directive
 argbash_api([INCLUDE_PARSING_CODE], _CHECK_PASSED_ARGS_COUNT(1, 2)[m4_do(
 	[[$0($@)]],
 	[m4_ifndef([SCRIPT_DIR_DEFINED], [m4_fatal([You have to define a script directory by some means before using '$0'])])],
 	[m4_list_append([_OTHER],
-		m4_expand([[. "$]m4_default_quoted([$2], _SCRIPT_DIR_NAME)[/$1]"  [# '.' means 'source'
+		m4_expand([[# shellcheck source=SCRIPTDIR/$1]]),
+		m4_expand([[. "$][{]m4_default_quoted([$2], _SCRIPT_DIR_NAME)[}][/$1]"  [# '.' means 'source'
```

Comment:
```suggestion
		[# shellcheck source=SCRIPTDIR/$1],
		m4_expand([[. "${]m4_default_quoted([$2], _SCRIPT_DIR_NAME)[}/$1]"  [# '.' means 'source'
```
Your solution is correct, I have just simplified it:

- There is no macro there, so no need to expand anything. `$1` is expanded automatically.
- The circus in the second line is because of a macro that has to be expanded. You just want to modify an existing literal string, so you can just modify existing fragments.